src/main/java/Duke.java

```diff
@@ -1,10 +1,183 @@
+            case COMMAND_BYE:
+                System.out.println(" Bye-bye. Hope to see you again soon!");
+                shouldExit = true;
+                break;
+
+            case COMMAND_LIST:
+                if (taskList.isEmpty()) {
+                    System.out.println(" Uhh.. It's empty..");
+                } else {
+                    System.out.println(" Here are the tasks in your list: ");
+                    for (int i = 0; i < taskList.size(); i++) {
+                        Task task = taskList.get(i);
+                        System.out.println("\t" + (i + 1) + ". " + task.toString());
+                    }
+                }
+                break;
+
+            case COMMAND_DONE:
+                int doneTaskIndex = Integer.parseInt(description) - 1;
+                boolean isValidTaskIndex = doneTaskIndex < taskList.size() && doneTaskIndex >= 0;
+                if (!isValidTaskIndex) {
+                    doneTaskIndex++;
+                    System.out.println(" Uhh... There's no task numbered: " + doneTaskIndex);
+                    break;
+                }
+
+                Task taskToMark = taskList.get(doneTaskIndex);
+                if (taskToMark.getStatus()) {
+                    System.out.println(" Remember? You have completed this task already.");
+                } else {
+                    pendingTaskCount--;
+                    taskList.get(doneTaskIndex).markAsDone();
+                    if (pendingTaskCount == 0) {
+                        System.out.println(" Awesome!! You are all caught up :)");
+                    } else {
+                        System.out.println(" Awesome!! Just " + pendingTaskCount + " more task(s) to go!");
+                    }
+                }
+
+                System.out.println("\t" + taskList.get(doneTaskIndex).toString());
+                break;
+
+            case COMMAND_TODO:
+                if (inputArguments.length == REQUIRED_NUMBER_OF_ARGUMENT_FOR_TODO) {
+                    isValidDescription = !description.isBlank();
+                    if (isValidDescription) {
+                        Todo newTodo = new Todo(description);
+                        addTask(taskList, newTodo);
+                        break;
+                    }
+                }
+
+                System.out.println(" Failed to add new todo: check syntax and no blanks!");
+                break;
+
+            case COMMAND_DEADLINE:
+                if (inputArguments.length == REQUIRED_NUMBER_OF_ARGUMENT_FOR_DEADLINE) {
+                    String byOption = inputArguments[2];
+                    String byInformation = inputArguments[3];
+
+                    isValidDescription = !description.isBlank();
+                    isValidOption = byOption.equals("/by");
+                    isValidOptionInformation = !byInformation.isBlank();
+                    isValidCommand = isValidDescription && isValidOption && isValidOptionInformation;
```

Comment:
I like how you are very detailed in validating the inputs! 👍

Reply:
Thank you.